src/BlockTypes/Cart.php

```diff
@@ -105,6 +112,30 @@ protected function render( $attributes, $content ) {
 			$content = preg_replace( '/<div class="[a-zA-Z0-9_\- ]*wp-block-woocommerce-cart[a-zA-Z0-9_\- ]*">/mi', $inner_blocks_html, $content );
 			$content = $content . '</div>';
 		}
+
+		/**
+		 * Cart i3 added inner blocks for Order summary. We need to add them to Cart i2 templates.
+		 * The order needs to match the order in which these blocks were registered.
+		 */
+		$coupons_enabled                 = wc_coupons_enabled();
+		$order_summary_with_inner_blocks = '$0
+			<div data-block-name="woocommerce/cart-order-summary-heading-block" class="wp-block-woocommerce-cart-order-summary-heading-block"></div>
+			<div data-block-name="woocommerce/cart-order-summary-subtotal-block" class="wp-block-woocommerce-cart-order-summary-subtotal-block"></div>
+			<div data-block-name="woocommerce/cart-order-summary-fee-block" class="wp-block-woocommerce-cart-order-summary-fee-block"></div>
+			<div data-block-name="woocommerce/cart-order-summary-discount-block" class="wp-block-woocommerce-cart-order-summary-discount-block"></div>' .
+			( $coupons_enabled ? '<div data-block-name="woocommerce/cart-order-summary-coupon-form-block" class="wp-block-woocommerce-cart-order-summary-coupon-form-block"></div>' : '' ) .
```

Comment:
This shouldn't be here and shouldn't be needed at all, the logic inside `cart-order-summary-coupon-form/block.tsx` should be enough. The logic to render those divs should match the logic we have inside `defaultTemplate` of `cart-order-summary-block/edit.tsx`, so if we don't have conditions inside that array, we shouldn't have logic here.

Reply:
You can confirm that by inserting a V3 Cart and saving, the HTML from coupon form is always there, regardless of taxes being enabled or not.